src/families/bitcoin/wallet-btc/xpub.ts

```diff
@@ -440,18 +415,25 @@ class Xpub extends EventEmitter {
     account: number,
     index: number
   ) {
-    const lastTx = await this.storage.getLastTx({
-      account,
-      index,
-      confirmed: true,
-    });
+    let pendingTxs: TX[] = [];
+    let txs: TX[] = [];
+    let inserted = 0;
+    do {
+      const lastTx = await this.storage.getLastTx({
+        account,
+        index,
+        confirmed: true,
+      });
 
-    const txs = await this.explorer.getAddressTxsSinceLastTxBlock(
-      this.txsSyncArraySize,
-      { address, account, index },
-      lastTx
-    );
-    const inserted = await this.storage.appendTxs(txs);
+      txs = await this.explorer.getAddressTxsSinceLastTxBlock(
+        this.txsSyncArraySize,
+        { address, account, index },
+        lastTx
+      );
+      pendingTxs = pendingTxs.concat(txs.filter((tx) => !tx.block));
```

Comment:
no concat because the ledger explorer always return all the pending tx on every page.
and a regular explorer would probably just return the pending tx after it returned the confirmed txs

Reply:
however they would not have been added twice, storage is careful about that